### PR TITLE
Ensure only courses only show on dropdown/radios if they're open for applications

### PR DESCRIPTION
**Pull request by davidgisbey**
## Context

Publish gives providers the option to accept applications for a course from a specific date. Currently, we don't have the concept of courses opening on a specific date. In order to do this we're going to need to pull the date in as part of our sync.

This PR follows on from #5970 and ensures that only courses which are open for applications show in the dropdown/radio buttons on the pick course page.

It also ensures that only providers which have an open course, in the current cycle, which is open for applications show in the pick provider dropdown.

## Changes proposed in this pull request

- Update the pick provider dropdown to only have providers where a course is open for applications
- Update the pick course form/radio buttons to only have courses which are open for applications

## Link to Trello card

https://trello.com/b/dcWOMFyp/user-support-ticket-tracker

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/forms/candidate_interface/pick_course_form.rb [hunk 1]
```diff
@@ -8,12 +8,12 @@ class PickCourseForm
     DropdownOption = Struct.new(:id, :name)
 
     def radio_available_courses
-      @radio_available_courses ||= courses_for_current_cycle.exposed_in_find.order(:name).includes(:course_options)
+      @radio_available_courses ||= courses_for_current_cycle.exposed_in_find.where('courses.applications_open_from <= ?', Time.zone.today).order(:name).includes(:course_options)
```
Comment (TobyRet) [inline app/forms/candidate_interface/pick_course_form.rb:11]: https://github.com/DFE-Digital/apply-for-teacher-training/pull/5969#pullrequestreview-794426278

Comment (TobyRet) [reply app/forms/candidate_interface/pick_course_form.rb:11]: Looks like you can also extract the common query from line 67?

File: app/forms/candidate_interface/pick_course_form.rb [hunk 2]
```diff
@@ -64,7 +64,7 @@ def unique_name_for(course)
     end
 
     def available_courses
-      @available_courses ||= courses_for_current_cycle.exposed_in_find.includes(:accredited_provider, :course_options)
+      @available_courses ||= courses_for_current_cycle.exposed_in_find.where('courses.applications_open_from <= ?', Time.zone.today).includes(:accredited_provider, :course_options)
```
Comment (duncanjbrown) [inline app/forms/candidate_interface/pick_course_form.rb:67]: Should this be a scope too?

File: app/forms/candidate_interface/pick_course_form.rb (hunk 1 above)
Comment (davidgisbey) [reply app/forms/candidate_interface/pick_course_form.rb]: Great have done 👍

**Review by TobyRet — APPROVED** [review]
LGTM

**Pull request merged**